v2/event/event_marshal.go

```diff
@@ -179,10 +180,22 @@ func WriteJson(in *Event, writer io.Writer) error {
 		return fmt.Errorf("error while writing the event data: %w", stream.Error)
 	}
 
+	// Add extensions in a deterministic predictable order, similar to how Go maps are serialized in a predictable order.
+	type keyValue struct {
+		key string
+		val any
+	}
+	extensionKeyValues := make([]keyValue, 0, len(ext))
 	for k, v := range ext {
+		extensionKeyValues = append(extensionKeyValues, keyValue{k, v})
```

Comment:
Minor thing, but this makes a copy of "v", which if large, could be an issue. 
What do you think about just putting the ext names into a slice, sort it, then then iterate over the slice and grab the values from the ext map? Then the only copying is the ext key names.

Reply:
Thanks for the suggestion, that actually simplified the code nicely since I didn't need this `keyValue` pair type when I'm just working with the extension names.

I've updated the code accordingly